sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/MapListenerDelegate.kt

```diff
@@ -6,49 +6,91 @@ import com.mapbox.maps.plugin.delegates.listeners.*
  * Definition of a listener manager delegate to manage all kinds of listeners.
  */
 interface MapListenerDelegate {
+  // Camera Events
+  /**
+   * Add a listener that's going to be invoked whenever map camera changes.
+   */
+  fun addOnCameraChangeListener(onCameraChangeListener: OnCameraChangeListener)
 
   /**
-   * Add a listener that's going to be invoked whenever map state changes.
+   * Remove the camera change listener.
    */
-  fun addOnMapChangedListener(onMapChangeListener: OnMapChangedListener)
+  fun removeOnCameraChangeListener(onCameraChangeListener: OnCameraChangeListener)
 
+  // Map events
   /**
-   * Remove the map change listener.
+   * Add a listener that's going to be invoked whenever map has entered the idle state.
+   *
+   * The Map is in the idle state when there are no ongoing transitions and the Map has rendered all
+   * available tiles.
    */
-  fun removeOnMapChangedListener(onMapChangedListener: OnMapChangedListener)
+  fun addOnMapIdleListener(onMapIdleListener: OnMapIdleListener)
```

Comment:
@tobrun do I get it correctly that for majority of benchmarks now it's better to use that listener?

Reply:
If your benchmark needs to include symbol placement to be fully ready, yes you can use that dedicated listener.
if that doesn't need to be tracked, there are callbacks that called earlier.

Reply:
I see docs stating
```
  /**
   * Invoked when the Map has entered the idle state. The Map is in the idle state when there are no ongoing transitions
   * and the Map has rendered all available tiles.
   */
```
For me it seems like a classic convenient IDLE meaning that map has loaded everything (I see nothing here about symbol placement etc), no rendering will happen now and we could assume that benchmark could be ended. Thus it could be used literally everywhere as an exit point for a benchmark.
Am I wrong?

Reply:
@kiryldz The events is revised again in the gl-native beta.17 release, it will include more granular control over the loading status. I will integrate the changes to this PR when the gl-native release lands.

Reply:
I think I will move forward and merge this PR first, and adapt breaking changes again together with bumping gl-native in a separate PR. This PR is getting too big for review again.